src/SignalR/server/Core/src/Internal/TypedClientBuilder.cs

```diff
@@ -79,27 +85,23 @@ private static IEnumerable<MethodInfo> GetAllInterfaceMethods(Type interfaceType
             }
         }
 
-        private static void BuildConstructor(TypeBuilder type, FieldInfo proxyField)
+        private static ConstructorInfo BuildConstructor(TypeBuilder type, FieldInfo proxyField)
         {
-            var method = type.DefineMethod(".ctor", System.Reflection.MethodAttributes.Public | System.Reflection.MethodAttributes.HideBySig);
-
-            var ctor = typeof(object).GetConstructor(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic,
-                null, new Type[] { }, null);
+            var ctor = type.DefineConstructor(MethodAttributes.Public, CallingConventions.Standard, ParameterTypes);
 
-            method.SetReturnType(typeof(void));
-            method.SetParameters(typeof(IClientProxy));
-
-            var generator = method.GetILGenerator();
+            var generator = ctor.GetILGenerator();
 
             // Call object constructor
             generator.Emit(OpCodes.Ldarg_0);
-            generator.Emit(OpCodes.Call, ctor);
+            generator.Emit(OpCodes.Call, typeof(object).GetConstructors().Single());
```

Comment:
Nit: Should we cache the object ConstructorInfo?

Reply:
Hmm. I don't know where though. Storing it in a static field would still result in one per `T`, no? 🤔

Reply:
I guess it's the same for the existing `CancellationTokenNoneProperty` and the new `ParameterTypes` as well.